### ISSUE TITLE
for-each.sh doesn't support multiple commands

### ISSUE THREAD

**Issue by zepumph** [phetsims/perennial#136]
Often I have wanted to do something like:

`for-each.sh perennial/data/active-sims npm install && grunt update`

but currently it won't work.

After quite a bit of experimenting (some with @samreid and @chrisklus), I think I found a solution.

If we change this line:
`    (cd ${repository} > /dev/null; "${@:2}") # command is all args after the filename`
to
`    (cd ${repository} > /dev/null; eval "${@:2}") # command is all args after the filename`

Then we evaluate it as a string so both of the following commands work:

With quotes:
`for-each.sh perennial/data/active-sims "npm install && grunt update"`

With no quotes:
`for-each.sh perennial/data/active-sims grunt update`

I'm going to commit this, because it seems like an improvement and I can't see any issues with it. I'll get someone more fluent in bash to review the change to make sure there is no edge case I didn't think of.

**Comment by zepumph** [comment]
@mattpen will you please review. Does this make sense? I don't know too much about `eval` as a command.

**Comment by zepumph** [comment]
Once done it may be nice to share this at dev meeting as a PSA. I'm unsure how many others have run into this.

**Comment by mattpen** [comment]
> @mattpen will you please review. Does this make sense? I don't know too much about eval as a command.

I have never used eval and I'm not aware of the problems.  I reviewed http://mywiki.wooledge.org/BashFAQ/048 and it sounds like the main problem with eval is that it can execute arbitrary code if it is passed information from a user.  However, in our case this sounds like desired behavior.  At the end of that article there are some guidelines for robust usage.

In the past, when wanting to use for-each.sh on a more complicated command I've just written a new script with a for loop or some kind of xargs command.  Is this needed for something other than a one-time-use type of script?

**Comment by zepumph** [comment]
>In the past, when wanting to use for-each.sh on a more complicated command I've just written a new script with a for loop or some kind of xargs command.

That is not something I am easily able to write, and I bet that is the same for many phet devs.

>Is this needed for something other than a one-time-use type of script?

Yes I am thinking that it would be used as "one-off"ish as any `for-each.sh` script I run, like running a grunt/node/npm/git task once. 

I looked at the "robust usages" section, and it seemed like most of it was overkill for the purposes of `for-each.sh`. Thanks for you your help. We haven't heard any complaints about this script breaking for devs in the last 10 days, and I've been using the added functionality too without issue. Closing
